geth/api/backend.go

```diff
@@ -77,120 +80,146 @@ func (m *StatusBackend) IsNodeRunning() bool {
 }
 
 // StartNode start Status node, fails if node is already started
-func (m *StatusBackend) StartNode(config *params.NodeConfig) (<-chan struct{}, error) {
-	m.Lock()
-	defer m.Unlock()
-
-	if m.nodeReady != nil {
-		return nil, node.ErrNodeExists
-	}
+func (m *StatusBackend) StartNode(config *params.NodeConfig) error {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	return m.startNode(config)
+}
 
-	nodeStarted, err := m.nodeManager.StartNode(config)
+func (m *StatusBackend) startNode(config *params.NodeConfig) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("node crashed on start: %v", err)
+		}
+	}()
+	err = m.nodeManager.StartNode(config)
 	if err != nil {
-		return nil, err
+		switch err.(type) {
+		case node.RPCClientError:
```

Comment:
To avoid repetition, it is possible to only figure out `Event.Error` value and have a single `signal.Send()` all after the `switch`.

Reply:
done